lib/grammers-client/src/client/auth.rs

```diff
@@ -14,25 +14,51 @@ use grammers_tl_types as tl;
 use std::convert::TryInto;
 use std::fmt;
 
+use grammers_crypto::two_factor_auth::{calculate_2fa, VerificationError};
+
 /// The error type which is returned when signing in fails.
 #[derive(Debug)]
 pub enum SignInError {
-    SignUpRequired {
-        terms_of_service: Option<tl::types::help::TermsOfService>,
-    },
     InvalidCode,
+    IncorrectParameters(VerificationError),
+    IncorrectPassword,
```

Comment:
Above we're using `Invalid` instead of `Incorrect` code. Perhaps it would make sense to stick to the same naming convention?:

```suggestion
    InvalidParameters(VerificationError),
    InvalidPassword,
```